ruff_sync.py

```diff
@@ -94,6 +94,26 @@ def _get_cli_parser() -> ArgumentParser:
     return parser
 
 
+def github_url_to_raw_url(url: URL) -> URL:
+    """
+    Convert a GitHub URL to its corresponding raw content URL
+
+    Args:
+        url (URL): The GitHub URL to be converted.
+
+    Returns:
+        URL: The corresponding raw content URL.
+    """
+    url_str = str(url)
+    if "github.com" in url_str and "/blob/" in url_str:
+        raw_url_str = url_str.replace("github.com", "raw.githubusercontent.com").replace(
+            "/blob/", "/"
+        )
+        return httpx.URL(raw_url_str)
+    else:
+        return url
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (edge_case_not_handled):** Consider handling more GitHub URL variants.

The current implementation assumes a specific format for GitHub URLs. Consider handling other common variants, such as URLs containing 'gist.github.com' or those using 'https://'.

```suggestion
def github_url_to_raw_url(url: URL) -> URL:
    url_str = str(url)
    if "github.com" in url_str and "/blob/" in url_str:
        raw_url_str = url_str.replace("github.com", "raw.githubusercontent.com").replace("/blob/", "/")
    elif "gist.github.com" in url_str:
        raw_url_str = url_str.replace("gist.github.com", "gist.githubusercontent.com").replace("/raw/", "/")
    else:
        return url
    return httpx.URL(raw_url_str)
```